Eplant/views/ChromosomeViewer/Viewer/GeneList.tsx

```diff
@@ -0,0 +1,247 @@
+	useSetGeneticElements,
+	useSidebarState
+} from "@eplant/state"
+import CloseIcon from '@mui/icons-material/Close';
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import DialogTitle from "@mui/material/DialogTitle";
+import IconButton from "@mui/material/IconButton";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from "@mui/material/ListItemText";
+import Popover from "@mui/material/Popover";
+import useTheme from "@mui/material/styles/useTheme";
+import Typography from "@mui/material/Typography";
+
+import { GeneIcon } from "../icons";
+import { GeneArray, GeneItem } from "../types";
+
+// TYPES
+interface GeneListProps {
+	id: string,
+	start: number,
+	end: number,
+	anchorOrigin: Array<number>
+}
+
+
+//----------
+// COMPONENT
+//----------
+const GeneList: FC<GeneListProps> = ({
+	id,
+	start,
+	end,
+	anchorOrigin
+}) => {
+
+
+	const [loading, setLoading] = useState<boolean>(true);
+	// gene list
+	const [geneList, setGeneList] = useState<GeneArray>([{
+		id: "",
+		start: 0,
+		end: 0,
+		strand: "",
+		aliases: [],
+		annotation: ""
+	}]);
+	const [selectedIndex, setSelectedIndex] = useState<number | null>(null);
+	const [selectedGene, setSelectedGene] = useState<GeneItem | null>(null)
+	// gene info popup
+	const [open, setOpen] = useState(false);
+
+	// Other/Global State
+	const [activeGeneId, setActiveGeneId] = useActiveGeneId()
+	const geneticElements = useGeneticElements()
+	const setGeneticElements = useSetGeneticElements()
+	const collections = useCollections()
+	const setCollections = useSetCollections()
+	const theme = useTheme()
+
+
+
+	//------------------
+	// Helper Functions
+	//------------------
+	React.useEffect(() => {
+		const fetchGeneData = async () => {
+			const response: Response = await fetch(
+				`https://bar.utoronto.ca/eplant/cgi-bin/querygenesbyposition.cgi?chromosome=${id}&start=${start}&end=${end}`
+			);
+			// await sleep(1000)
+			if (response.ok) {
+				const geneListData: GeneArray = await response.json();
+				setGeneList(geneListData);
+
+			}
+
+		};
+		fetchGeneData();
+		setLoading(false)
+	}, []);
+	// EVENT HANDLERS
+	const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+		setOpen(true)
+	}
+	const handleGeneSelect = (gene: GeneItem, index: number) => (event: React.MouseEvent<HTMLElement>) => {
+		setSelectedIndex(index)
+		setSelectedGene(gene)
+		console.log("Clicked Gene: ", gene)
+	}
+	const handleClose = () => {
+		setOpen(false)
+	}
+	const handleLoadGeneClick = (event: React.MouseEvent<HTMLElement>) => {
+		if (selectedGene != null) {
+			const gene = new GeneticElement(
+				selectedGene.id,
+				selectedGene.annotation,
+				arabidopsis,
+				selectedGene.aliases
+			)
+			geneticElements[0].push(gene)
+			setGeneticElements(geneticElements[0])
```

Comment:
```suggestion
			setGeneticElements([...geneticElements, gene])
```
The reason your state change wasn't immediately triggering a re-render is because React checks for [referential equality](https://blog.bitsrc.io/understanding-referential-equality-in-react-a8fb3769be0) to listen for state change. 

In general it is good to view state objects as immutable and on any update to state, create a new object with the reflected changes. In this case you just need to create a new array copy with the need geneticElement appended to it. @zkdan Correct me if I made any errors here.